src/channel.rs

```diff
@@ -0,0 +1,160 @@
+            } else if pre_str.starts_with("rc") {
+                Ok(Channel::RC)
+            } else {
+                Err(ChannelError::FailedToParseVersion(value))
+            }
+        } else {
+            Ok(Channel::Stable)
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::Channel;
+    use semver::{AlphaNumeric, Version};
+    use std::convert::TryFrom;
+
+    #[test]
+    fn alpha_version() {
+        let v = Version {
+            major: 1,
+            minor: 1,
+            patch: 0,
+            pre: vec![AlphaNumeric("alpha01".to_owned())],
+            build: vec![],
+        };
+        let channel = Channel::try_from(v);
+        assert_eq!(Channel::Alpha, channel.unwrap());
+    }
+
+    #[test]
+    fn beta_version() {
+        let v = Version {
+            major: 1,
+            minor: 1,
+            patch: 0,
+            pre: vec![AlphaNumeric("beta01".to_owned())],
+            build: vec![],
+        };
+        let channel = Channel::try_from(v);
+        assert_eq!(Channel::Beta, channel.unwrap());
+    }
+
+    #[test]
+    fn dev_version() {
+        let v = Version {
+            major: 1,
+            minor: 1,
+            patch: 0,
+            pre: vec![AlphaNumeric("dev01".to_owned())],
+            build: vec![],
+        };
+        let channel = Channel::try_from(v);
+        assert_eq!(Channel::Dev, channel.unwrap());
+    }
+
+    #[test]
+    fn rc_version() {
+        let v = Version {
+            major: 1,
+            minor: 1,
+            patch: 0,
+            pre: vec![AlphaNumeric("rc01".to_owned())],
+            build: vec![],
+        };
+        let channel = Channel::try_from(v);
+        assert_eq!(Channel::RC, channel.unwrap());
+    }
+
+    #[test]
+    fn stable_version() {
+        let v = Version::new(1, 1, 1);
+        let channel = Channel::try_from(v);
+        assert_eq!(Channel::Stable, channel.unwrap());
+    }
+
+    #[test]
+    fn cmp_channels() {
+        assert!(Channel::Stable > Channel::RC);
+        assert!(Channel::RC > Channel::Beta);
+        assert!(Channel::Beta > Channel::Alpha);
+        assert!(Channel::Alpha > Channel::Dev);
+    }
+
+    #[test]
+    fn cmp_parsed_versions() {
+        let stable = Channel::try_from(Version::parse("1.1.0").unwrap()).unwrap();
+        let rc = Channel::try_from(Version::parse("1.1.0-rc01").unwrap()).unwrap();
+        let alpha = Channel::try_from(Version::parse("1.1.0-alpha01").unwrap()).unwrap();
+        let beta = Channel::try_from(Version::parse("1.1.0-beta01").unwrap()).unwrap();
+        assert!(Channel::Stable >= stable);
+        assert!(Channel::Stable >= rc);
+        assert!(Channel::Stable >= alpha);
+        assert!(Channel::Stable >= beta);
+        assert!(Channel::RC >= rc);
+        assert!(Channel::RC >= beta);
+        assert!(Channel::RC >= alpha);
+        assert!(Channel::Beta >= beta);
+        assert!(Channel::Beta >= alpha);
+        assert!(Channel::Alpha >= alpha);
+    }
```

Comment:
~~Tests for any potential regression in `std::cmp::PartialOrd`~~ who am I kidding I wrote this to actually understand my own logic.